nibabel/wrapstruct.py

```diff
@@ -536,8 +536,18 @@ def __str__(self):
 
         def _getter(obj, key):
             try:
-                return obj.get_value_label(key)
+                val = obj.get_value_label(key)
             except ValueError:
-                return obj[key]
+                val = obj[key]
+            # Convert a dtype('S1') numpy string to a Python str, if possible
+            if isinstance(val, np.ndarray) and val.dtype.type == np.bytes_:
+                try:
+                    tmp = val.tolist().decode()
+                    # Do not return unicode objects (Python 2)
+                    if isinstance(tmp, str):
```

Comment:
The issue the code was trying to solve is that pretty-printing `np.asarray(b'xyz')` (i.e. `'%s' % np.asarray(b'xyz')` results in `xyz` in Python 2 and `b'xyz'` in Python 3.

The test used to isolate the poorly behaved Python 3 case also isolates well-behaved Python 2, and could result in a `u'xyz'`, which prints as `xyz` in Python 3 and `u'xyz'` in Python 2. However, if the `decode` fails, I want it to still print something, so return the `bytes` anyway.

The strategy used here is intended to avoid explicitly depending on Python version. There may be a better way to do this, and I'm open to suggestions. For example, returning `np.asarray(val.tolist().decode())` would work fine, because `'%s' % np.asarray(u'xyz')` behaves the same in both 2 and 3.

Let me know what you think.